lib/saml_config.rb

```diff
@@ -2,12 +2,10 @@
 
 module Saml
   class Config
-    def initialize
-      @@settings ||= build_settings.freeze
-    end
+    attr_reader :settings
 
-    def settings
-      @@settings
+    def initialize
+      @settings ||= build_settings.freeze
```

Comment:
are we sure that there's a difference in terms of re-reading from disk? I have been googling around it and seems like the only difference between instance vars and class vars is scope  availability http://stackoverflow.com/questions/15773552/ruby-class-instance-variable-vs-class-variable